ui/components/multichain/account-list-item/account-list-item.js

```diff
@@ -115,17 +116,19 @@ const AccountListItem = ({
   const [accountListItemMenuElement, setAccountListItemMenuElement] =
     useState();
 
-  const snap = useSelector((state) =>
-    getSnap(state, account.metadata?.snap?.id),
-  );
   const snapMetadata = useSelector(getSnapsMetadata);
-  const accountLabel = getAccountLabel(
-    account.metadata.keyring.type,
-    account,
-    account.metadata.keyring.type === KeyringType.snap
-      ? getSnapName(snapMetadata)(account.metadata?.snap?.id)
-      : null,
-    snap?.preinstalled,
+  const keyrings = useSelector(getMetaMaskKeyrings);
+  const accountLabels = useMemo(
+    () =>
+      getAccountLabels(
+        account.metadata.keyring.type,
+        account,
+        keyrings,
+        account.metadata.keyring.type === KeyringType.snap
+          ? getSnapName(snapMetadata)(account.metadata?.snap.id)
```

Comment:
In case this is really `undefined` for some reason, we don't want this to throw (and the original code was already using this):

```suggestion
          ? getSnapName(snapMetadata)(account.metadata?.snap?.id)
```